src/types/router.ts

```diff
@@ -14,19 +15,26 @@ export type RouterPushOptions = {
   replace?: boolean,
 }
 
-type RouterPushRoute<
-  TRoutes extends Routes,
-  TRoute extends PropertyKey
-> = RouterPushOptions & {
-  name: TRoute,
-  params?: TRoute extends keyof Flattened<TRoutes> ? Flattened<TRoutes>[TRoute] : Record<never, never>,
-}
+type RoutePushConfigParams<
```

Comment:
This type just makes sure the `params` property is not required if there are no required params.